### ISSUE TITLE
Eloquent model silently omits fields when using "firstOrNew" on a model with fillable defined

### ISSUE THREAD

**Issue by john-mark-wilkinson** [laravel/framework#8658]
If you request `firstOrNew` on an eloquent model, where there are `fillable` properties defined, if no model is found, one will be returned with only the properties defined in `fillable`.

So if I have the properties `foreign_key` and `type` which together create a candidate key, and I only have `type` in my `fillable` array, a new model object will be returned, with only `type` defined.

This problem could either be fixed in a number of ways. 

1. Having eloquent fail when attempting this
2. Having it ignore `fillable` on `firstOrNew`, and set all the submitted properties
3. Documenting this behavior.

**Comment by JosephSilber** [comment]
I believe this to be expected behavior.

If you want force fill it, use the `unguarded` method:

```php
$attributes = [...];

$model = User::unguarded(function() use ($attributes)
{
    return User::firstOrNew($attributes);
});
```

**Comment by lukasgeiter** [comment]
I agree with Joseph. The mass assignment protection wouldn't make sense if it wasn't always enabled.

**Comment by john-mark-wilkinson** [comment]
If this is expected and desired behavior (which I think is fine), it needs to be documented.

**Comment by lasselehtinen** [comment]
I actually bumped into this same thing. I had earlier defined one fillable attribute and I was wondering why it does not assign the rest of them. For me it would make much more sense if firstOrCreate and firstOrNew would throw MassAssignmentException like Model::create does in this case. We are still creating a new model based on array aren't we?

**Comment by lukasgeiter** [comment]
@lasselehtinen `Model::create()` behaves exactly the same as `firstOrCreate()` or `firstOrNew()`. If at least one `$fillable` attributes is defined no exception will be thrown. The simple reason being that they all use the constructor and then `fill()` to assign the attributes.

**Comment by lasselehtinen** [comment]
@lukasgeiter OK I did not actually know this. This might be a stupid question, but why shouldn't it throw a MassAssignmentException if you try to pass other attributes that are not fillable? Is bit for cases when user can pass for example garbage parameters from the URL and you pass it on to Model::create(Input::all()) and you don't want to throw an exception in those cases by default? In that scenario this behavior makes sense, I agree.

But like @john-mark-wilkinson mentioned, it would be probably wise to mention this in the documentation. I checked the WIP 5.1 documentation and seems like Taylor hasn't finished the Eloquent part just yet.
